python/cuml/manifold/t_sne.pyx

```diff
@@ -339,11 +346,34 @@ class TSNE(Base):
                           "# of datapoints = {}.".format(self.perplexity, n))
             self.perplexity = n
 
-        # Prepare output embeddings
-        Y = rmm.device_array(
-            (n, self.n_components),
-            order="F",
-            dtype=np.float32)
+        # PCA Intialization
```

Comment:
IMO, should be doing this in C++ and put PCA initialization as an option to make it easier for users in that layer. This would be more consistent with the way other algorithms (like UMAP) work. It will also make it just as easy for user of the C++ API to call T-SNE as users of the Python API

Reply:
Yep good point I'll change it!
The only issue I had was the input is row-major, whilst if I remember correctly cuML on the C++ layer supports column-major.

I could edit cov.h and add row-major support

Reply:
Hmm. You shouldn't have to change cov.h to support row-major for this. Since cuDF is the more natural input to our algorithms you just need to change the `input_utils` in the Python layer to use Fortran layout. The transpose will need to be done either way, but you can do it after the PCA init in C++.

Reply:
@cjnolet Ohh thats a fantastic idea!!!!!!! Essentially ask for F contiguous layout then inside C++ call MLCommon::LinAlg::transpose(X) to make it C contiguous!